rs/sns/swap/src/swap.rs

```diff
@@ -3472,6 +3362,144 @@ fn create_sns_neuron_basket_for_neurons_fund_participant(
     Ok(recipes)
 }
 
+enum ConversionError {
+    Invalid,
+    Skip,
+}
+
+impl SnsNeuronRecipe {
+    /// Converts a SnsNeuronRecipe (a Swap concept) to an SnsNeuronRecipe (an SNS Governance concept)
+    fn to_neuron_recipe(
```

Comment:
The more I look into this function, the more I think it should just be a `TryFrom` with auxiliary data.